### DISCUSSION TITLE
How to be informed when strings need translation?

### DISCUSSION THREAD

**Discussion by Patriccollu** [sumatrapdfreader/sumatrapdf#3949] [Q&A] [answered]
Hello,

I'm Corsican translator for SumatraPDF and I use [apptranslator.org](https://www.apptranslator.org/app/SumatraPDF).

Is there a way to be informed when there are new or modified strings that need translation?

Until recently, each translated language had its specific file that contained all the strings : for example `translations/co.txt` for Corsican. So it was easy to use the GitHub's Atom feed to be informed when there was a commit made to that file. But now all these language specific files do no longer exist.

As `translations/translations.txt` file contains both the original and the translated strings (_for all the supported languages_), it is regularly modified to add new strings _but also to add various translations_. So when there is a commit, this doesn't want to say that new strings are added.

Do you know another way to reach this goal?

Thanks,
Patriccollu.

**Comment by kjk** [comment]
Hmm, didn't think of this. Currently there is no better way than watching `translations.txt`

**Comment by Patriccollu** [comment]
When there are new strings, I understand that `translations.txt` is updated but I guess there is also _an action_ to populate [apptranslator.org](https://www.apptranslator.org/app/SumatraPDF). Would it be possible to implement _some additional process_ at that moment?

As an example, creating _English_ as language would allow to monitor this RSS feed:
https://www.apptranslator.org/rss?app=SumatraPDF&lang=en

**Comment by Patriccollu** [comment]
@kjk 
Do you think that my previous suggestion would be easy to implement?

Happy New Year 2024 to the whole team!

**Comment by Patriccollu** [comment]
@kjk 
Is there another file which is always updated at the same time as `translations.txt` ?
Could it be possible to create English as language in [apptranslator.org](https://www.apptranslator.org/app/SumatraPDF) which would allow to monitor the following RSS feed:
https://www.apptranslator.org/rss?app=SumatraPDF&lang=en

**Comment by kjk** [comment]
@Patriccollu I've added the count on untranslated strings to RSS feed for the language. If new string gets added then untranslated count will change which should show up in RSS reader.

That's the theory, at least. Haven't tested it yet.

**Comment by Patriccollu** [comment]
@kjk
Thank you for taking my request in consideration.

I was already monitoring this RSS feed before - https://www.apptranslator.org/rss?app=SumatraPDF&lang=co - and was informed when there were translations modified in Corsican... but not when there were new string added.

I will tell you if the 'new' RSS feed - with the untranslated count - will cover my need. Thanks again.

**Reply by kjk** [reply, marked as answer]
It did work for me. I've made 2 changes to rss feed for each language:
* added `<N> untranslated in language...` which will change if I add new string
* changed `updated` attribute of rss entry to be the newer of "last translation in language" or "last added untranslated string" (previously it was just "last translation in language")

So your rss reader should see those changes. It might take some time for the reader to notice. I use feedly and it took over a day to notice the changes but they did show.

**Comment by Patriccollu** [comment]
Hello @kjk 

Hurray! This morning I received an email - via the Corsican RSS feed - https://www.apptranslator.org/rss?app=SumatraPDF&lang=co - which informed me that there were 3 untranslated strings in language `co`.

For your information, I have received an email on February 29<sup>th</sup> and March 3<sup>rd</sup> displaying `0 untranslated in language co` and I can't explain the reason because there was no changes with the previous situation.

But I can confirm that this 'new' RSS feed - with the untranslated count - perfectly covers my need!

So I'm closing this topic.

Thanks,
Patriccollu.
